app/submitter/submitter.py

```diff
@@ -66,15 +83,17 @@ def send_message(self, message, queue, tx_id):
         try:
             connection = self._connect()
             channel = connection.channel()
-            channel.queue_declare(queue=queue, durable=True)
+
+            channel.queue_declare(queue=self.queue, durable=True)
             properties = BasicProperties(headers={},
                                          delivery_mode=2)
 
+            tx_id = kwargs['tx_id']
```

Comment:
if this is a `kwarg` do we need to check its existence?

Reply:
or use `kwargs.get('tx_id')`

Reply:
Can do, only didn't because it's passed in using `.get()`